src/alchemlyb/estimators/ti_gaussian_quadrature_.py

```diff
@@ -0,0 +1,220 @@
+            ad_delta += np.diagflat(np.array(dout), k=j)
+
+        # yield standard delta_f_ cumulative free energies from one state to another
+        self._delta_f_ = pd.DataFrame(
+            adelta, columns=means_new.index.values, index=means_new.index.values
+        )
+        
+        # yield standard deviation d_delta_f_ between each state
+        self._d_delta_f_ = pd.DataFrame(
+            np.sqrt(ad_delta),
+            columns=variances_new.index.values,
+            index=variances_new.index.values,
+        )
+        
+        self.dhdl = means
+        self.dhdl.attrs = dHdl.attrs
+        self._states_ = means_new.index.values.tolist()
+        self._delta_f_.attrs = dHdl.attrs
+        self._d_delta_f_.attrs = dHdl.attrs
+
+        return self
+
+    @staticmethod
+    def separate_mean_variance(means, variances):
+        """
+        For transitions with multiple lambda, the attr:`dhdl` would return
+        a :class:`~pandas.DataFrame` which gives the dHdl for all the lambda
+        states, regardless of whether it is perturbed or not. This function
+        creates 3 lists of :class:`numpy.array`, :class:`pandas.Series` and 
+        :class:`pandas.Series` for each lambda, where the lists describe 
+        the lambda values, potential energy gradient and variance values for
+        the lambdas state that is perturbed.
+
+        Parameters
+        ----------
+        means: DataFrame
+            means is the average potential energy gradient at each lambda.
+        variances: DataFrame
+            variances is variance of the potential energy gradient at each lambda.
+
+        Returns
+        ----------
+        lambda_list : list
+            A list of :class:`numpy.array` such that ``lambda_list[k]`` is the
+            lambda values with respect to each type of lambda.
+        dhdl_list : list
+            A list of :class:`pandas.Series` such that ``dHdl_list[k]`` is the
+            potential energy gradient with respect to lambda for each
+            configuration that lambda k is perturbed.
+        variance_list : list
+            A list of :class:`pandas.Series` such that ``variance_list[k]`` is the
+            variance of the potential energy gradient with respect to lambda for each
+            configuration that lambda k is perturbed.
+        """
+        lambda_list = []
+        dhdl_list = []
+        variance_list = []
+        # get the lambda names
+        l_types = means.index.names
+        # get the lambda vaules
+        lambdas = means.reset_index()[means.index.names].values
+
+        if len(means.index.names) == 1:
+            name = means.columns[0]
+            lambda_list.append(means.index)
+            dhdl_list.append(means[name])
+            variance_list.append(variances[name])
+
+        else:
+            # simultanouesly scaling of multiple lambda types are not supported
+            if (((0.0 < lambdas) & (lambdas < 1.0)).sum(axis=1) > 1.0).any():
```

Comment:
I'm a bit confused as to when this would be triggered. So I would imagine that for lambda `coul` and `vdw`.
You would want to accept

|coul|vdw|
|-|---|
|0|0.1|
|0|0.2|
|0|...|
|0|0.9|

Or

|coul|vdw|
|-|---|
|1|0.1|
|1|0.2|
|1|...|
|1|0.9|

Or

|coul|vdw|
|-|---|
|0.1|0.1|
|0.2|0.2|
|...|...|
|0.9|0.9|

And you want to reject

|coul|vdw|
|-|---|
|0.2|0.1|
|0.5|0.2|
|...|...|
|1|0.9|
?

Reply:
@xiki-tempula Thanks for the comment. Could I double check with you if the coul and vdw columns are independent from each other? For example, if we have a table of the means like:

| coul-lambda | vdw-lambda | coul-dHdl-means | vdw-dHdl-means |
| ------------  | ------------  | -----------  | -----------  | 
| 0.0469 | 0.0469 | 25.71 | -12.65 |   
| 0.2308 |  0.2308 | 17.67 | -3.11 |
| 0.5 | 0.5 | 8.77 | 3.15 |
| 0.7692 | 0.7692 | 3.17 | 5.27 |
| 0.9531 | 0.9531 | 0.56 | 5.36 |

Could we do the estimation separately? Like:
`∆G_coul = W_0.0469 * 25.71 + W_0.2308 * 17.67  + W_0.5 * 8.77 + W_0.7692 * 3.17 + W_0.9531 * 0.56`
`∆G_vdw = W_0.0469 * (-12.65) + W_0.2308 * (-3.11)  + W_0.5 * 3.15 + W_0.7692 * 5.27 + W_0.9531 * 5.36 `
`∆G = ∆G_coul + ∆G_vdw`

If so, perhaps we don't need this check.

Reply:
> Could I double check with you if the coul and vdw columns are independent from each other? 

For Gromacs, this is the case. If you check the TI code then it just add them together. I think this check might not be necessary.

Reply:
Okay, sounds good. I will remove it then.